build-logic/build.gradle.kts

```diff
@@ -61,7 +61,9 @@ java {
 }
 
 tasks.withType<JavaCompile>().configureEach {
-  options.release.set(17)
+  // Do not use options.release - see https://issuetracker.google.com/issues/278800528
+  sourceCompatibility = "17"
+  targetCompatibility = "17"
```

Comment:
IIRC, this is only for Android compile tasks. Java compile tasks should still use `options.release.set(17)`

Reply:
FWIW, this is how I do it in librarian: https://github.com/GradleUp/librarian/blob/2be199a3b4f3d3cac8f7560d101a138d64ae8f96/librarian-gradle-plugin/src/main/kotlin/com/gradleup/librarian/gradle/compatibility.kt#L50

This is definitely not tested against 2.1.0 so take with large grains of salt